src/infrasys/h5_time_series_storage.py

```diff
@@ -214,51 +328,88 @@ def get_time_series(
             normalization=metadata.normalization,
         )
 
+    @_get_time_series_dispatch.register(DeterministicMetadata)
+    def _(
+        self,
+        metadata: DeterministicMetadata,
+        start_time: Optional[datetime] = None,
+        length: Optional[int] = None,
+        context: Any = None,
+    ) -> DeterministicTimeSeriesType:
+        """Return a DeterministicTimeSeries array."""
+        assert context is not None
+
+        root = context[self.HDF5_TS_ROOT_PATH]
+        uuid = str(metadata.time_series_uuid)
+
+        if uuid not in root:
+            msg = f"Time series with {uuid=} not found"
+            raise ISNotStored(msg)
+
+        dataset = root[uuid]["data"]
+
+        # index, length = metadata.get_range(start_time=start_time, length=length)
+        # #DeterministicMetadata does not have get_range
+        data = dataset[:]  # Get all data
```

Comment:
Shouldn't we be able to read one window?

Reply:
I left that out of this PR since we I do not foresee someone using this in a near future.